lib/java/ydoc-polyfill/src/main/java/org/enso/ydoc/polyfill/WebSocketClient.java

```diff
@@ -0,0 +1,112 @@
+package org.enso.ydoc.polyfill;
+
+import java.io.File;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Stream;
+import org.enso.ydoc.polyfill.web.WebEnvironment;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.HostAccess;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+
+/**
+ * Helper main class to execute provided JavaScript file with the {@code org.enso.ydoc.polyfill}
+ * emulation. The emulation exposes <a href="http://nodejs.org">node.js</a> compatible environment
+ * with <a href="https://nodejs.org/en/learn/getting-started/websocket">WebSocket</a> and <a
+ * href="https://github.com/websockets/ws">WebSocketServer</a> objects being emulated. The goal of
+ * the emulation isn't 100% compatible replacement (yet), but (only) to execute enough of <a
+ * href="https://github.com/yjs/yjs">Y.js</a> code for Enso purposes.
+ *
+ * <h3>Usage</h3>
+ *
+ * Get built version of Enso engine distribution. Or use <code>sbt buildEngineDistribution</code> to
+ * build it yourself from the Enso sources. There is {@code component} directory in the
+ * installation. Use the GraalVM's Java command to execute:
+ *
+ * <pre>
+ * $ graalvm/bin/java
+ *    -p enso-engine-*?/enso-*?/component/
+ *    -m org.enso.ydoc.polyfill/org.enso.ydoc.polyfill.WebSocketClient
+ * </pre>
+ *
+ * The {@code enso-engine-*?/enso-*?/component/} represents path to the @{code component} directory
+ * which needs to be adjusted to proper path on each operating system. All lines may need to be
+ * concatenated into a single line.
```

Comment:
The `WebSocketClient` should be a useful utility for testing compatibility of the emulation with real node.js implementation.